handlers/anteHandler.go

```diff
@@ -96,61 +95,37 @@ func NewAnteHandler(utxoStore store.UTXOStore, plasmaStore store.PlasmaStore, cl
 func validateInput(ctx sdk.Context, input plasma.Input, signer common.Address, utxoStore store.UTXOStore, client plasmaConn) (*big.Int, sdk.Result) {
 	var amt *big.Int
 
-	if input.IsDeposit() {
-		deposit, ok := client.GetDeposit(input.DepositNonce)
-		if !ok {
-			return nil, msgs.ErrInvalidTransaction(DefaultCodespace, "deposit, %s, does not exist", input.DepositNonce.String()).Result()
-		}
-
-		// add deposit to app state if non existent
-		if !utxoStore.HasUTXO(ctx, deposit.Owner, input.Position) {
-			utxo := store.UTXO{
-				Output:   plasma.NewOutput(deposit.Owner, deposit.Amount),
-				Position: input.Position,
-				Spent:    false,
-			}
-
-			utxoStore.StoreUTXO(ctx, utxo)
-		}
-
-		// the owner of the deposit might not equal the signer so we must explicity
-		// check for a match
-
-		if !bytes.Equal(deposit.Owner[:], signer[:]) {
-			return nil, sdk.ErrUnauthorized(fmt.Sprintf("signer does not own the deposit: Signer: %x, Owner: %x", signer, deposit.Owner)).Result()
-		}
-
-		amt = deposit.Amount
-	} else {
-		// inputUTXO must be owned by the signer due to the prefix so we do not need to
-		// check the owner of the position
-		inputUTXO, ok := utxoStore.GetUTXO(ctx, signer, input.Position)
-		if !ok {
-			return nil, msgs.ErrInvalidTransaction(DefaultCodespace, "input, %s, does not exist by owner %x", inputUTXO.Position, signer).Result()
-		}
-		if inputUTXO.Spent {
-			return nil, msgs.ErrInvalidTransaction(DefaultCodespace, "input, %s, already spent", inputUTXO.Position).Result()
-		}
+	// inputUTXO must be owned by the signer due to the prefix so we do not need to
+	// check the owner of the position
+	inputUTXO, ok := utxoStore.GetUTXO(ctx, signer, input.Position)
+	if !ok {
+		return nil, msgs.ErrInvalidTransaction(DefaultCodespace, "input, %s, does not exist by owner %x", inputUTXO.Position, signer).Result()
```

Comment:
Change inputUTXO.Position to input.Position. This is the err msg in #64 . inputUTXO is returned as nil causing strange err to be returned